### PR TITLE
Fixed cmtx nonce

### PR DESCRIPTION
**Pull request by lyh169**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okexchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: libs/cosmos-sdk/baseapp/baseapp_checktx.go [hunk 1]
```diff
@@ -68,3 +75,13 @@ func (app *BaseApp) CheckTx(req abci.RequestCheckTx) abci.ResponseCheckTx {
 		Events:      info.result.Events.ToABCIEvents(),
 	}
 }
+
+// for adaptivet the same sender multi-tx in mempool can add to TxQueue
+func (app *BaseApp) updataCheckTxResponseNonce(tx sdk.Tx, mode runTxMode, senderNonce uint64) {
```
Comment (giskook) [inline libs/cosmos-sdk/baseapp/baseapp_checktx.go]:
```suggestion
func (app *BaseApp) updateCheckTxResponseNonce(tx sdk.Tx, mode runTxMode, senderNonce uint64) {
```

File: app/app.go [hunk 2]
```diff
@@ -1011,3 +1012,12 @@ func NewEvmSysContractAddressHandler(ak *evm.Keeper) sdk.EvmSysContractAddressHa
 		return ak.IsMatchSysContractAddress(ctx, addr)
 	}
 }
+
+func NewUpdateCMTxNonceHandler() sdk.UpdateCMTxNonceHandler {
+	return func(tx sdk.Tx, nonce uint64) {
+		stdtx, ok := tx.(*authtypes.StdTx)
+		if ok && nonce != 0 {
```
Comment (giskook) [inline app/app.go:1019]: Maybe we should limit the nonce to be positive
```suggestion
		if ok && nonce > 0 {
```

Comment (lyh169) [reply app/app.go:1019]: it's uint64, so no need >0

File: libs/cosmos-sdk/baseapp/baseapp_checktx.go (hunk 1 above)
Comment (lyh169) [reply libs/cosmos-sdk/baseapp/baseapp_checktx.go]: ok

**Review by ilovers — DISMISSED** [review]

**Review by giskook — DISMISSED** [review]

**Review by xzav3r — APPROVED** [review]

**Review by ilovers — APPROVED** [review]

**Review by giskook — APPROVED** [review]

**Pull request merged**